src/libarena/lib.rs

```diff
@@ -201,11 +201,8 @@ impl<T> TypedArena<T> {
                     return;
                 } else {
                     new_capacity = last_chunk.storage.cap();
-                    loop {
+                    while new_capacity < currently_used_cap + n {
```

Comment:
Isn't there a subtle difference in behaviour here?

Namely, previous code would always execute at least one iteration, after your change it is not necessarily true.

Reply:
There is, yes, but this approach doesn't increase the capacity in case it is already sufficient for growth, so I thought it was preferable.

Reply:
`grow` is never called when there already is sufficient capacity, so this just adds an extra check.